src/actions/upload_actions.js

```diff
@@ -47,7 +48,8 @@ export default class UploadActions {
       return null;
     }
     const timestamp = await this.blockchainStateWrapper.getBlockTimestamp(uploadBlock);
-    const {transactionHash} = await this.uploadsWrapper.getUploadData(bundleId, uploadBlock);
+    const events = await this.challengesEventEmitterWrapper.challenges(uploadBlock, uploadBlock);
+    const [{transactionHash}] = events.filter(({returnedValues}) => returnedValues.bundleId === bundleId);
```

Comment:
You could use here `find` instead of `filter` which would stop on the first match.